### PR TITLE
feat: Add Plugin V3

### PR DESCRIPTION
**Pull request by yevgenypats**

### CODE REVIEW

**Review by erezrokah — APPROVED** [review]

File: plugin/v3/plugin.proto [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+enum Operator {
+  EQUAL = 0;
+  LESS_THAN = 1;
+}
+
+enum PK_MODE {
+  DEFAULT = 0;
+  CQ_ID_ONLY = 1;
+}
```
Comment (hermanschaaf) [inline plugin/v3/plugin.proto:44]:
```suggestion
enum Registry {
  REGISTRY_UNSPECIFIED = 0;
  REGISTRY_GITHUB = 1;
  REGISTRY_GRPC = 2;
  REGISTRY_LOCAL = 3;
}

enum Scheduler {
  SCHEDULER_UNSPECIFIED = 0;
  SCHEDULER_DFS = 1;
  SCHEDULER_ROUND_ROBIN = 2;
}

enum PKMode {
  PK_MODE_UNSPECIFIED = 0;
  PK_MODE_DEFAULT = 0;
  PK_MODE_CQ_ID_ONLY = 1;
}
```

Comment (hermanschaaf) [reply plugin/v3/plugin.proto:44]: Couple of suggestions:
 - Use Pascal case for enum names (follows the style guide: https://protobuf.dev/programming-guides/style/#enums)
 - Add an "_UNSPECIFIED" as the first entry in each case so it's possible to distinguish unset values
 - Removes unused enums

File: plugin/v3/plugin.proto [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      MessageInsert insert = 2;
+      MessageDeleteStale delete = 3;
+    }
+  }
+}
+
+message Write {
+  message Request {
+    oneof message {
+      WriteOptions options = 1;
```
Comment (hermanschaaf) [inline plugin/v3/plugin.proto:129]: Should `options` not be outside of the oneof?

File: plugin/v3/plugin.proto [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+message WriteOptions {
+  bool migrate_force = 1;
+}
+
+message MessageMigrateTable {
+  // marshalled arrow.Schema
+  bytes table = 1;
+  bool migrate_force = 2;
+}
```
Comment (hermanschaaf) [inline plugin/v3/plugin.proto:80]: We now have migrate_force in two places: which one should be preferred?

Comment (yevgenypats) [reply plugin/v3/plugin.proto:81]: removed this one.

File: plugin/v3/plugin.proto (hunk 1 above)
Comment (yevgenypats) [reply plugin/v3/plugin.proto]: addressed.

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**